experiments/basic_experiment_runner.py

```diff
@@ -216,7 +220,7 @@ def validate_learned_domain(
         self.export_learned_domain(
             learned_model,
             domains_backup_dir_path,
-            f"{self._learning_algorithm.name}_fold_{fold_number}_{learned_model.name}" f"_{len(allowed_observations)}_trajectories.pddl",
+            f"{learning_algorithm_name}_fold_{fold_number}_{learned_model.name}" f"_{len(allowed_observations)}_trajectories.pddl",
```

Comment:
No. Don't change the function's signature, this will affect the other experiments as well and we will have to make sure that the entire code is compatible. 
Furthermore, why did you make this change?

Reply:
The self._learning_algorithm.name attribute is set to ma_sam during the creation of the multi_agent_experiment and remains unchanged thereafter. In each iteration, both sam and masam call the export_learned_domain method and use this value, causing one to override the other's files as they are named the same.

To address this issue, I decided to pass the value from the outside. The caller of this function knows which algorithm is currently in use, allowing it to pass this value. 

You are right in that it affects other places, this function is used in one other place.

Reply:
An alternative approach is to update this self._learning_algorithm.name side-effect variable in each iteration, similar to how other side effects are managed. For instance, when learning the sam_model, we use the following lines:

  self.domain_validator.learning_algorithm = LearningAlgorithmType.sam_learning
  self.learning_statistics_manager.learning_algorithm = LearningAlgorithmType.sam_learning

And these are changed back to 'ma_sam' when ma_sam is learned.
Repeated each iteration.

We could investigate whether it is safe to apply the same logic to self._learning_algorithm.name during sam and masam learning.


multi_agent_experiment structurally lacks cohesion wise, as it is dependent on values of ma_sam, yet it learns both sam and masam.

So another approach is to try to to separate the experiment to more cohesive models.

Reply:
I'll add an issue to set it as a parameter for the function so that it would be cleaner and will not be set as a side effect, for now this is good.